pkg/transport/generic_transport_controller.go

```diff
@@ -847,8 +852,19 @@ func (c *genericTransportController) computeDestToCustomizedObjects(objectsToPro
 	return destToCustomizedObjects, bindingErrors
 }
 
+func TODO_FIXME_ADD_CONSTANT_CREATEONLY(obj *unstructured.Unstructured) Wrapee {
+	return Wrapee{obj, false}
+}
+
 func (c *genericTransportController) wrapBatch(batchToPropagate []*unstructured.Unstructured, binding *v1alpha1.Binding, numShard int, isSharded bool) (*unstructured.Unstructured, error) {
-	wrappedObject, err := convertObjectToUnstructured(c.transport.WrapObjects(batchToPropagate))
+	var wrapped runtime.Object
+	if t2, is := c.transport.(TransportWithCreateOnly); is {
```

Comment:
Does this not beat the point of "generic" in `genericTransportController`? Shouldn't the difference in functionality be abstracted in the interface and implemented twice?

Due to the urgency (if i'm not mistaken) - it is fine to keep it as-is but add a TODO or document this in any way - that is assuming you agree.

Reply:
Resolved